src/main.go

```diff
@@ -19,6 +19,12 @@ func main() {
 	}
 	log.Info("initialization of server completed")
 
+	// Regenerate NGINX templates on restart
+	repoURL, err := controllers.NginxRegenerate()
+	if err != nil {
+		log.Errorf("Could not regenerate NGINX template for repo: %s, details: %v", repoURL, err)
```

Comment:
Put this before initialization of server and do a Falatf if there is an error. We do not want to start the server if there was an error in the nginx entries setup the first time.